src/screens/WalletConnect/CallRequest/SignatureRequestContent.js

```diff
@@ -47,10 +54,19 @@ type Props = {|
 function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
   const { t } = useTranslation();
   const configs = useChainsConfig();
+  const activeAccount = useActiveAccount();
+  const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
 
   const { title, iconUrl, chain } = getViewData(request);
   const config = configs[chain];
 
+  /**
+   * Archanova account needs to be deployed regardless of the action,
+   * Etherspot account must to be deployed for signature requests only
+   */
+  const requiresDeployedAccount = (isArchanovaAccount(activeAccount) && !isArchanovaAccountDeployed)
+    || isEtherspotAccountDeployed(activeAccount, chain);
```

Comment:
This condition still seems wrong for etherspot: 
* if etherspot account is deployed it will be true meaning it needs deployment
* if etherspot account is not deployed it will be false meaning it does not need deployment

* for archanova deployed/not-deployes it seems ok, but a bit hackish as it assumes that `isEtherspotAccountDeployed` will return false when account is not etherspot, which is not very clear.

Reply:
Whoops 👀